content/posts/backlogged-pinball.md

```diff
@@ -0,0 +1,99 @@
+
+## Demo overview
+
+
+
+Backlogged pinball was designed to demonstrate specific cloud products and
+benefits in a fun, interactive way.
+
+{{< figure src="/images/backlogged-pinball/backlogged-arch.png" width="600px" title="The pinball machine sends pubsub events to the cloud which are processed by Cloud Run and displayed on an interactive web app. Users can deploy their own Cloud Run service to process events and send pubsub messages back to the machine." >}}
+
+In its current iteration, it highlights:
+
+1.  Cloud PubSub: Two-way PubSub communication with a novel source of events
+    (The pinball machine)
+1.  Cloud Run / Cloud functions: Both serverless platforms process pinball
+    events in real-time using different subscription mechanics (push vs. pull).
+1.  Cloud Run Deployments: Users can update a Cloud Run service in 20 seconds
+    which has an immediate effect on the current pinball game
+1.  Cloud Firestore: Firestore pushes all events to a live web-app. The round
+    trip from:
+
+    Pinball machine **->** PubSub **->** Cloud Functions **->** Firestore **->**
+    Web App
+
+    takes about 200 ms, and could probably be further optimized.
+
+## The machine
+
+{{< figure src="/images/backlogged-pinball/pinball_wide.jpeg" width="600px" title="The demo featured live-updating high scores and current game status" >}}
+
+The actual physical pinball machine is a
+[Multimorphic P3](https://www.multimorphic.com/p3-pinball-platform/). In short,
+this is a modular, programmable pinball machine with 2 large HD displays built
+in.
+
+Multimorphic also distributes
+[an SDK](https://www.multimorphic.com/p3-pinball-platform/3rd-party-developers/)
+to facilitate creating new pinball games.
+
+Thus, Backlogged is an application that uses the Unity game engine and the
+Multimorphic SDK to handle interfacing with the hardware (including displays).
+
+## What's next?
+{{< figure src="/images/backlogged-pinball/pinball_interview.jpeg" width="600px" title="Drew showing off the backlogged demo" >}}
+
+The pinball machine currently resides in a Google office in Salt Lake City.  Whenever I am able to, I plan on adding new features/functionality to the demo (with a major focus on cloud code).
+
+I am currently working on a game advisor that uses [llama3.1-70b](https://ollama.com/library/llama3.1:70b) to grade your game and recommend more optimal strategies. 
+
+{{< figure src="/images/backlogged-pinball/genkit.png" width="600px" title="A sample game analysis from the prototype" >}}
+
+In the future, I'd love to add more features like:
+
+*  Using AI to play the game and optimize for a high score
+*  Connecting two machines to the cloud to demonstrate more interactive
+    scenarios
+    *   This scenario could include two machines in different locations to
+        highlight network latency and resiliency
+*  More interactive features beyond simple messages and emoji.
+
+Any ideas for new features?  I'd love to hear them!
```

Comment:
how do folks respond to you? (not sure if there is a contact me somewhere)

Reply:
I don't have a good way yet... It's on my todo list. I don't want to expose my work (or really personal) email address, so I'm probably going to write some sort of feedback form that forwards emails to both.  And then document that as a blog post 😂

For now, it's just a treasure hunt for highly motivated readers to find my GitHub or LinkedIn accounts!